src/createServer.js

```diff
@@ -1,3 +1,54 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+/* eslint-disable max-len */
+'use strict';
+
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  return http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const textToConvert = normalizedUrl.pathname.slice(1);
+    const toCase = normalizedUrl.searchParams.get('toCase');
+    const caseOptions = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+    const errors = [];
+
+    if (!textToConvert) {
+      errors.push({
+        message: 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message: '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (toCase && !caseOptions.includes(toCase)) {
+      errors.push({
+        message: 'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    res.setHeader('Content-Type', 'application/json');
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const result = convertToCase(textToConvert, toCase);
+
+    result.targetCase = toCase;
+    result.originalText = textToConvert;
```

Comment:
Looks cooler this way :sunglasses: 
```suggestion
    const result = {
      ...convertToCase(textToConvert, toCase),
      targetCase: toCase,
      originalText: textToConvert
    };
```